lib/zclient.c

```diff
@@ -2434,36 +2434,44 @@ size_t zebra_interface_link_params_write(struct stream *s,
 	size_t w;
 	struct if_link_params *iflp;
 	int i;
+	struct in_addr inaddr_none_ip;
+
+	inaddr_none_ip.s_addr = INADDR_NONE;
 
-	if (s == NULL || ifp == NULL || ifp->link_params == NULL)
+	if (s == NULL || ifp == NULL)
 		return 0;
 
 	iflp = ifp->link_params;
+
 	w = 0;
 
-	w += stream_putl(s, iflp->lp_status);
+	w += stream_putc(s, iflp ? iflp->enabled : 0);
+	w += stream_putl(s, iflp ? iflp->lp_status : 0);
 
-	w += stream_putl(s, iflp->te_metric);
-	w += stream_putf(s, iflp->max_bw);
-	w += stream_putf(s, iflp->max_rsv_bw);
+	w += stream_putl(s, iflp ? iflp->te_metric : 0);
+	w += stream_putf(s, iflp ? iflp->max_bw : 0);
+	w += stream_putf(s, iflp ? iflp->max_rsv_bw : 0);
 
```

Comment:
does that mean bandwidth 0 will be sent with 0 ?
today, BW is set with enable, even if we do not set it.

I mean, are we sure that OSPF or ISIS will really remove the TLV when reading that value ?

Reply:
see the commit logs